benchto-driver/src/main/java/com/teradata/benchto/driver/listeners/measurements/ThroughputMeasurementProvider.java

```diff
@@ -29,12 +30,17 @@
         implements PostExecutionMeasurementProvider
 {
     @Override
-    public List<Measurement> loadMeasurements(Measurable measurable)
+    public CompletableFuture<List<Measurement>> loadMeasurements(Measurable measurable)
     {
+        List<Measurement> measurements;
         if (measurable instanceof BenchmarkExecutionResult && measurable.getBenchmark().isConcurrent() && measurable.isSuccessful()) {
-            return ImmutableList.of(measurement("throughput", "QUERY_PER_SECOND", calculateThroughput((BenchmarkExecutionResult) measurable)));
+            measurements = ImmutableList.of(measurement("throughput", "QUERY_PER_SECOND", calculateThroughput((BenchmarkExecutionResult) measurable)));
         }
-        return emptyList();
+        else {
+            measurements = emptyList();
+        }
+
+        return CompletableFuture.completedFuture(measurements);
```

Comment:
If I understand correctly, this code is still serial. What is the benefits of using `Future`-based API here?